x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1419,6 +1419,20 @@ static void validateForFips(Settings settings) {
                     + " ] setting."
             );
         }
+        final var cacheHashAlgoSettings = settings.filter(k -> k.endsWith("cache.hash_algo"));
```

Comment:
Nit: does it make sense to have a bit more strick check?
```suggestion
        final var cacheHashAlgoSettings = settings.filter(k -> k.endsWith(".cache.hash_algo"));
```

We have cache called `doc_cache`, but it does not have a `hash_algro` field. So it is not really an issue. But I wonder whether stricter is better?

Reply:
Yup, good point!